gadopt/time_stepper.py

```diff
@@ -377,16 +326,18 @@ def __init__(
         solver_parameters: Optional[dict[str, Any]] = {},
         strong_bcs: Optional[list[firedrake.DirichletBC]] = None,
     ):
-        # Get the Butcher tableau and stage type for this scheme
-        butcher, stage_type = gadopt_to_irksome_tableau(self.__class__.__bases__[1])
+        if self.butcher_tableau is None:
+            raise NotImplementedError(
+                f"{self.__class__.__name__} must define a butcher_tableau class attribute"
+            )
```

Comment:
Given [Python docs](https://docs.python.org/3/library/exceptions.html#NotImplementedError), you should probably raise `ValueError` here instead. Or you could make `butcher_tableau` an abstract method.